packages/app/src/sandbox/compile.ts

```diff
@@ -701,17 +713,31 @@ async function compile(opts: CompileOptions) {
           // The HTML is loaded from the server as a static file, no need to set the innerHTML of the body
           // on the first run. However, if there's no server to provide the static file (in the case of a local server
           // or sandpack), then do it anyways.
-          document.body.innerHTML = body;
+          if (body !== lastBodyHTML) {
+            document.body.innerHTML = body;
+
+            const scripts = await extractScripts(body, true);
+            if (scripts.length) {
+              // Reload page to re-execute scripts properly...
+              if (!firstLoad) {
+                window.location.reload();
+                return;
+              }
+              appendScripts(scripts, document.body);
+            }
+          }
 
           // Add head tags or anything that comes from the template
           // This way, title and other meta tags will overwrite whatever the bundler <head> tag has.
           // At this point, the original head was parsed and the files loaded / preloaded.
+          if (head !== lastHeadHTML) {
+            document.head.innerHTML += head;
```

Comment:
This will only get called on first load of local/sandpack environment so I think this is fine